src/plots/cartesian/axes.js

```diff
@@ -956,10 +954,14 @@ axes.tickFirst = function(ax) {
     var sRound = axrev ? Math.floor : Math.ceil;
     // add a tiny extra bit to make sure we get ticks
     // that may have been rounded out
-    var r0 = rng[0] * 1.0001 - rng[1] * 0.0001;
+    var r0 = expandRange(rng)[0];
     var dtick = ax.dtick;
     var tick0 = r2l(ax.tick0);
 
+    if(ax.tickmode === 'auto' && ax.rangebreaks && ax.maskBreaks(tick0) === BADNUM) {
+        tick0 = moveToEndOfBreak(tick0, ax);
```

Comment:
This still bothers me - I see it's only in `auto` mode but it seems like this will still have strange effects, especially if you do something unusual with the bounds of the break, like `[8.75, 17.25]`. That said I don't understand how we're getting the results we are in `axes_breaks-dtick_auto` of ticks at 9am and noon, ie alternating 3-hour and 5-hour gaps. If we're hitting this block then `tick0` gets set to 9am, but then to get a tick at noon we'd need a 3-hour dtick, which would then put another tick at 3pm - not what we see. And I don't think we want a 3/5 alternation as an autotick result anyway, we should regularize that to consistent 4-hour spacing.

So a couple of image tests, like you have, are good. But I think we also want to see a bunch of jasmine tests where we just set up an axis and list the ticks (positions and labels) that result within one day. That should let us cover a whole lot more cases: auto-determined dtick from 1 hour to a day and everything between, and a few manually set dtick and tick0 values; each of these tested against each of the hourly rangebreaks I mentioned in https://github.com/plotly/plotly.js/issues/4722#issuecomment-624892053